version: '3.5'

services:
  geoserver-master:
    networks:
      - traefik-net
    deploy:
      mode: replicated
      replicas: 1
      labels:
        traefik.port: "${GEOSERVER_PORT}"
        traefik.docker.network: traefik-net
        traefik.frontend.rule: "Host:${GEOSERVER_SUBDOMAIN}.${PUBLIC_HOSTNAME};PathPrefix:/geoserver,PathPrefix:/geoserver/web,PathPrefix:/geoserver/rest"
        traefik.frontend.priority: "1"
        traefik.backend: geoserver-master
      placement:
        constraints:
          - node.role == worker
      restart_policy:
        condition: on-failure
        max_attempts: 3

  geoserver-slave:
    networks:
      - traefik-net
    deploy:
      mode: replicated
      replicas: 2
      labels:
        traefik.port: "${GEOSERVER_PORT}"
        traefik.docker.network: traefik-net
        traefik.frontend.rule: "Host:${GEOSERVER_SUBDOMAIN}.${PUBLIC_HOSTNAME};PathPrefix:/geoserver/{((?!web).*)};PathPrefix:/geoserver/{((?!rest).*)}"
        traefik.frontend.priority: "1"
        traefik.backend: geoserver-slave
      placement:
        constraints:
          - node.role == worker
      restart_policy:
        condition: on-failure
        max_attempts: 3

networks:
  traefik-net:
    external: true

volumes:
  geoserver-vol:
    name: "geoserver-vol"
    driver: "cloudstor:aws"
    driver_opts:
      backing: shared
